helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -283,6 +287,68 @@ public void reset() {
     }
   }
 
+  /**
+   * Loads className using classLoader
+   * @param classLoader
+   * @param className
+   * @return Class className loaded by classLoader
+   */
+  private Class loadClass(URLClassLoader classLoader, String className) {
+    try {
+      return classLoader.loadClass(className);
+    } catch (ClassNotFoundException e) {
+      LOG.error("Failed to load Task class " + className + " for new task in instance " + _manager
+          .getInstanceName() + " in cluster " + _manager.getClusterName() + ".");
+      throw new IllegalStateException("Class not found for task");
+    }
+  }
+
+  /**
+   * Loads Task and TaskFactory classes for command input from
+   * a JAR file, and registers the TaskFactory in _taskFactoryRegistry.
+   * @param command The command indicating what task to be loaded
+   */
+  private void loadNewTask(String command) {
+    // If the path for dynamic tasks doesn't exist, skip loading the task
+    if (!_manager.getHelixDataAccessor().getBaseDataAccessor()
+        .exists(TaskConstants.DYNAMICALLY_LOADED_TASK_PATH, 0)) {
+      return;
+    }
+
+    // Read DynamicTaskConfig containing task definition information.
+    DynamicTaskConfig taskConfig = new DynamicTaskConfig(
+        _manager.getHelixDataAccessor().getBaseDataAccessor()
+            .get(TaskConstants.DYNAMICALLY_LOADED_TASK_PATH + "/" + command, null, 0));
```

Comment:
Instead of 0 (a magic number), please use a static constant that's already defined in the library.

Reply:
Does it make sense to define constant for Path separator "/"?  Not sure if it is defined anywhere else in helix